cmd/influx/cli/cli.go

```diff
@@ -1178,8 +1182,29 @@ func (c *CommandLine) ExecuteFluxQuery(query string) error {
 			}
 		}()
 	}
-	// TODO(lesam): steal 2.x flux client
-	return fmt.Errorf("ERROR: flux repl missing due to flux upgrade")
+	request := fluxClient.QueryRequest{}.WithDefaults()
+	request.Query = query
+	request.Dialect.Annotations = csv2.DefaultDialect().Annotations
+
+	results, err := c.Client.QueryFlux(ctx, &request)
+	if err != nil {
+		return err
+	}
+	for results.More() {
+		res := results.Next()
+		fmt.Fprintln(w, "Result:", res.Name())
+
+		if err := res.Tables().Do(func(tbl flux.Table) error {
+			_, err := newFormatter(tbl).WriteTo(w)
+			return err
+		}); err != nil {
+			return err
+		}
+	}
+	// It is safe and appropriate to call Release multiple times and must be
+	// called before checking the error on the next line.
+	results.Release()
```

Comment:
Should we `defer` this call right after the err-check for `QueryFlux`? Looks like we aren't releasing if we hit errors while iterating through results.

Reply:
Oops, yes we should. In fact I thought I had. Good catch!